blockchain/common_test.go

```diff
@@ -0,0 +1,129 @@
+package blockchain_test
```

Comment:
What was your reasoning for naming this package `blockchain_test`?

Reply:
keeping this as part of blockchain package causes import cycle
So I took reference from fullblocks_test and example_test files which are part of the package blockchain_test
https://github.com/utreexo/utreexod/blob/7e21d3d7ba843a7559e771091c373b88dc7dc07b/blockchain/fullblocks_test.go#L6

Reply:
2 things:

1. In Go, the package name must match the directory the file is in. So for a package `blockchain_test`, you'd have to make a new directory.
2. I'd like to avoid new packages and I don't think the tests being added here really need a new package. If import cycles are a problem, you could the tests into package fullblocktests. You'd have to bring back the HeightByHash function for this.

Reply:
If we take the second approach, then we won't need to export testGenerator as well, so we can remove that commit
But we need to export ChainSetup(change chainSetup to ChainSetup) from blockchain package.

Reply:
Also should I put the fullblocks_test and example_test files which exists already as part of blockchain_test package in fullblockstest package?

Reply:
> If we take the second approach, then we won't need to export testGenerator as well, so we can remove that commit
But we need to export ChainSetup(change chainSetup to ChainSetup) from blockchain package.

Sounds good. Should do the exporting in a separate commit as well.

> Also should I put the fullblocks_test and example_test files which exists already as part of blockchain_test package in fullblockstest package?

Hmm ok I did some searching and apparently having a `_test` package is used at times. Let's keep the existing tests where they are for now and just move `AcceptHeader` and `RejectHeader` tests into fullblocktests package since these rely on the TestGenerator a lot.